io/include/traccc/io/event_data.hpp

```diff
@@ -0,0 +1,96 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/cell.hpp"
+#include "traccc/edm/cluster.hpp"
+#include "traccc/edm/measurement.hpp"
+#include "traccc/edm/particle.hpp"
+#include "traccc/edm/spacepoint.hpp"
+#include "traccc/edm/track_candidate.hpp"
+#include "traccc/geometry/silicon_detector_description.hpp"
+#include "traccc/io/csv/cell.hpp"
+#include "traccc/io/csv/hit.hpp"
+#include "traccc/io/csv/measurement.hpp"
+#include "traccc/io/csv/measurement_hit_id.hpp"
+#include "traccc/io/csv/particle.hpp"
+#include "traccc/utils/seed_generator.hpp"
+
+// Detray include(s).
+#include "detray/core/detector.hpp"
+#include "detray/io/frontend/detector_reader.hpp"
+
+// Vecmem include(s).
+#include <vecmem/memory/memory_resource.hpp>
+
+// System include(s).
+#include <map>
+#include <string>
+
+namespace traccc {
+
+struct event_data {
+
+    public:
+    // Type definitions
+    using detector_type = detray::detector<detray::default_metadata,
+                                           detray::host_container_types>;
+
+    event_data() = delete;
+
+    event_data(const std::string& event_dir, const std::size_t event_id,
+               vecmem::memory_resource& resource, const detector_type& det,
+               const silicon_detector_description::host* dd);
+
+    void fill_cca_result(
+        const cluster_container_types::host& clusters,
+        const measurement_collection_types::host& measurements);
+
+    track_candidate_container_types::host generate_truth_candidates(
+        seed_generator<detector_type>& sg, vecmem::memory_resource& resource);
+
+    // IO EDM containers
+    std::vector<io::csv::cell> m_cells;
+    std::vector<io::csv::hit> m_hits;
+    std::vector<io::csv::measurement> m_measurements;
+    std::vector<io::csv::measurement_hit_id> m_meas_hit_ids;
+    std::vector<io::csv::particle> m_particles;
```

Comment:
I don't like using the CSV types like this. :frowning: This way the code can literally only work with a CSV input. Could you not implement the logic with only the EDM types that sit in `traccc::core`?

Which of course also brings up the question of why this class is not rather sitting in `traccc::performance` instead of `traccc::io`. :thinking:

Reply:
event_data is moved to performance directory. The csv containers are removed
But I am still using `io::csv::cell` in many places instead of struggling with SoA data type (even with the help of proxy it was a nightmare for me), where I need your mercy..